File: navigation/src/main/java/com/example/navigation/stepsEngine/field/IsReservationField.java
package com.example.navigation.stepsEngine.field;

import com.example.navigation.stepsEngine.payment.FlowState;

public class IsReservationField extends Field {

    @Override
    public Boolean get(FlowState flowState) {
        return flowState.getReservation().isReservation();
    }

    @Override
    public String getId() {
        return null;
    }

    @Override
    public void set(FlowState flowState, Object value) {

    }

}

File: navigation/src/main/java/com/example/navigation/stepsEngine/field/IsReservationField.kt
package com.example.navigation.stepsEngine.field

import com.example.navigation.stepsEngine.payment.FlowState

class IsReservationField : Field() {
    override operator fun get(flowState: FlowState): Boolean? {
        return flowState.reservation.isReservation
    }

    override fun getId(): String {
        return FieldName.IS_RESERVATION
    }

    override operator fun set(flowState: FlowState, value: Any?) {}
}